mesheryctl/internal/cli/root/registry/update.go

```diff
@@ -66,29 +91,41 @@ mesheryctl registry update --spreadsheet-id 1DZHnzxYWOlJ69Oguz4LkRVTFM79kC2tuvdw
 		if err != nil {
 			return ErrUpdateRegistry(err, modelLocation)
 		}
+		utils.Log.Debugf("Logger created and set debug log level at %s", logFilePath)
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-
-		srv, err := mutils.NewSheetSRV(spreadsheeetCred)
+		info, err := os.Stat(modelLocation)
 		if err != nil {
-			utils.Log.Error(ErrUpdateRegistry(err, modelLocation))
-			return err
+			return fmt.Errorf("invalid --input path '%s': %w", modelLocation, err)
 		}
-		resp, err := srv.Spreadsheets.Get(spreadsheeetID).Fields().Do()
-		if err != nil || resp.HTTPStatusCode != 200 {
-			utils.Log.Error(ErrUpdateRegistry(err, outputLocation))
-			return err
+		if !info.IsDir() {
+			return fmt.Errorf("--input path '%s' is not a directory", modelLocation)
 		}
 
-		sheetGID = GetSheetIDFromTitle(resp, "Components")
+		utils.Log.Debugf("Input Directory check completed with path  %s", modelLocation)
 
-		err = InvokeCompUpdate()
+		var parser ComponentSourceParser
+
+		if csvDir != "" {
+			utils.Log.Info("Using local CSV directory: ", csvDir)
+			parser = &LocalCSVDirParser{dirPath: csvDir}
+		} else {
+			utils.Log.Info("Using Google Sheet with ID: ", spreadsheeetID)
+			parser = &GoogleSheetParser{spreadsheeetID: spreadsheeetID, spreadsheeetCred: spreadsheeetCred}
+		}
```

Comment:
```suggestion
		}
		if {
			utils.Log.Info("Using Google Sheet with ID: ", spreadsheeetID)
			parser = &GoogleSheetParser{spreadsheeetID: spreadsheeetID, spreadsheeetCred: spreadsheeetCred}
		}
```

you can let it as it is now but but I feel it is a little more readable

Reply:
since the requirement lets csv local takes precedence when both local dir and google sheet provides, I put the check like that. But I can reorder the conditions for readability then.